core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/data/impl/DataFeedServiceImpl.java

```diff
@@ -189,14 +188,27 @@ public void deleteDataFeed() throws DataFeedException {
             removePreviousDataFeed(serviceResourceResolver);
             removeCsvReport(serviceResourceResolver);
             removePendingNode(serviceResourceResolver);
-            gridResourcesCache.clearCache();
+            clearStaticDataFeed();
             serviceResourceResolver.commit();
         } catch (PersistenceException e) {
             LOG.error("Failed to delete data feed", e);
             throw new DataFeedException("Exception Deleting Data Feed");
         }
     }
 
+    private List<GridResource> getGridResourcesList() {
+        return new ArrayList<>(GRID_RESOURCE_COPY_ON_WRITE_ARRAY_LIST);
+    }
+
+    private synchronized void setGridResourcesList(List<GridResource> gridResources) {
+        GRID_RESOURCE_COPY_ON_WRITE_ARRAY_LIST.clear();
+        GRID_RESOURCE_COPY_ON_WRITE_ARRAY_LIST.addAll(gridResources);
+    }
+
+    private synchronized void clearStaticDataFeed() {
```

Comment:
In in this case, I believe, we do not need nethier a synchronized class method not a synchronized block becase `clear()` is an atomic operation, and our CopyOnWriteList is thread-safe already